### ISSUE TITLE
CLI not sending errors in threads with `showInThreads` enabled

### ISSUE THREAD

**Issue by deepanchal** [ryanrosello-og/playwright-slack-report#80]
I am using the following config to send slack reports:
```json
{
  "sendResults": "always",
  "slackLogLevel": "debug",
  "sendUsingBot": {
    "channels": ["qa"]
  },
  "showInThread": true,
  "meta": [
    { "key": "name", "value" : "__ENV_REPORT_NAME"},
    { "key": "build", "value" : "__ENV_BUILD_ID"},
    { "key": "branch", "value" : "__ENV_BRANCH_NAME"},
    { "key": "commit", "value" : "__ENV_COMMIT_ID"},
    { "key": "results", "value" : "__ENV_RESULT_URL"}
  ],
  "maxNumberOfFailures": 10,
  "disableUnfurl": true
}
```

Job Step:

```yaml
      - name: 📨 Send report to slack
        env:
          SLACK_BOT_USER_OAUTH_TOKEN: ${{ secrets.SLACK_BOT_USER_OAUTH_TOKEN }}
          REPORT_NAME: Playwright Report | ${{ matrix.stage }}
          BUILD_ID: ${{ github.run_id }}
          BRANCH_NAME: ${{ github.ref_name }}
          COMMIT_ID: ${{ github.sha }}
          # Todo: replace with html report url when we upload html reports somewhere
          RESULT_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
        run: |
          cat > cli_config.json <<EOF
          {
            "sendResults": "always",
            "slackLogLevel": "debug",
            "sendUsingBot": {
              "channels": ["qa"]
            },
            "showInThread": true,
            "meta": [
              { "key": "name", "value" : "__ENV_REPORT_NAME"},
              { "key": "build", "value" : "__ENV_BUILD_ID"},
              { "key": "branch", "value" : "__ENV_BRANCH_NAME"},
              { "key": "commit", "value" : "__ENV_COMMIT_ID"},
              { "key": "results", "value" : "__ENV_RESULT_URL"}
            ],
            "maxNumberOfFailures": 10,
            "disableUnfurl": true
          }
          EOF
          echo "Created cli_config.json for playwright-slack-report..."
          cat cli_config.json
          echo "Playwright JSON report:"
          cat e2e-results.json
          echo "Sending report to slack..."
          pnpm exec playwright-slack-report --config=cli_config.json --json-results=e2e-results.json
```

Full github workflow: https://github.com/CrisisCleanup/crisiscleanup-4-web/blob/fix/e2e-workflow-fixes-2/.github/workflows/e2e.yml

### Workflow with Option A from README

Workflow file: https://github.com/CrisisCleanup/crisiscleanup-4-web/actions/runs/7359923330/workflow

Workflow run with thread message: https://github.com/CrisisCleanup/crisiscleanup-4-web/actions/runs/7359923330/job/20035377215#step:13:743

![image](https://github.com/ryanrosello-og/playwright-slack-report/assets/46167867/fae21e91-8a12-4687-b7a1-2dbd4e471b55)

### Workflow with Option C from README

Workflow file: https://github.com/CrisisCleanup/crisiscleanup-4-web/actions/runs/7360054143/workflow

Workflow run without thread message: https://github.com/CrisisCleanup/crisiscleanup-4-web/actions/runs/7360054143/job/20035735669#step:8:2295

![image](https://github.com/ryanrosello-og/playwright-slack-report/assets/46167867/e690f4b5-e5f6-4d72-af91-6f7ea0b798a0)

**Comment by ryanrosello-og** [comment]
hey @deepanchal 
I can see in your config that you are using webhooks.  Unfortunately, the `showInThread` configuration is not supported by the Slack api when used with Webhooks.
https://github.com/ryanrosello-og/playwright-slack-report/issues/71#issuecomment-1856450006

**Comment by deepanchal** [comment]
@ryanrosello-og 
I am using `SLACK_BOT_USER_OAUTH_TOKEN` and 
```json
  "sendUsingBot": {
    "channels": ["qa"]
  },
  ```
in my config. I followed steps in readme to get `SLACK_BOT_USER_OAUTH_TOKEN` and it's not using webhooks. I am using Option C from readme to send slack messages with bot. I don't have any playwright-slack-report config in my playwright.config.ts. Can you please tell me where I am using webhooks?

**Comment by ryanrosello-og** [comment]
@deepanchal can you send the json file that you are trying to send

**Comment by deepanchal** [comment]
@ryanrosello-og Sure

cli_config.json
```json
{
  "sendResults": "always",
  "slackLogLevel": "debug",
  "sendUsingBot": {
    "channels": ["qa"]
  },
  "showInThread": true,
  "meta": [
    { "key": "name", "value" : "__ENV_REPORT_NAME"},
    { "key": "build", "value" : "__ENV_BUILD_ID"},
    { "key": "branch", "value" : "__ENV_BRANCH_NAME"},
    { "key": "commit", "value" : "__ENV_COMMIT_ID"},
    { "key": "results", "value" : "__ENV_RESULT_URL"}
  ],
  "maxNumberOfFailures": 10,
  "disableUnfurl": true
}
```

playwright json report: https://gist.github.com/deepanchal/fb977a5c6cd4e16fc6ffff2666975766

**Comment by ryanrosello-og** [comment]
@deepanchal this should be fixed now in v1.1.58

![2024-01-06_20-48-56](https://github.com/ryanrosello-og/playwright-slack-report/assets/50574915/73e72a42-28ef-448f-91ab-6ee7f90794cf)

**Comment by deepanchal** [comment]
@ryanrosello-og Thanks for fixing this bug. It's working as expected now

![image](https://github.com/ryanrosello-og/playwright-slack-report/assets/46167867/ab8800ed-943c-40c6-b957-6d5e2b231425)
